op-challenger/game/fault/types/types_test.go

```diff
@@ -3,10 +3,113 @@ package types
 import (
 	"math/big"
 	"testing"
+	"time"
 
+	"github.com/ethereum/go-ethereum/common"
 	"github.com/stretchr/testify/require"
 )
 
+func clockFromParts(duration, timestamp uint64) *Clock {
```

Comment:
This would be neater if the go Clock implementation wasn't so tightly coupled to the contract encoding too.